internal/testsdetection/testsdetection_test.go

```diff
@@ -0,0 +1,87 @@
+package testsdetection_test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testsdetection"
+)
+
+var coverDir string
+
+func TestMustBeTestingInTests(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		require.Nil(t, r, "MustBeTesting should not panic as we are running in tests")
+	}()
+
+	testsdetection.MustBeTesting()
+}
+
+func TestMustBeTestingInProcess(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		integrationtestsTag bool
+
+		wantPanic bool
+	}{
+		"Pass when called in an integration tests build": {integrationtestsTag: true, wantPanic: false},
+
+		"Error (panics) when called in non tests and no integration tests": {integrationtestsTag: false, wantPanic: true},
```

Comment:
Just a nit, but personally I find inline test case structs very hard to follow :stuck_out_tongue_closed_eyes:

Reply:
On the contrary, I find it way easier and it describes a story :)